presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -752,6 +754,13 @@ private PrestoSparkQueryExecution(
                         if (sparkException.getMessage().contains("most recent failure: JVM_OOM")) {
                             wrappedPrestoException = new PrestoException(SPARK_EXECUTOR_OOM, executionException);
                         }
+                        else if (sparkException.getMessage().matches(".*Total size of serialized results .* is bigger than allowed maxResultSize.*")) {
+                            wrappedPrestoException = new PrestoException(EXCEEDED_SPARK_DRIVER_MAX_RESULT_SIZE, executionException);
+                        }
+                        else if (sparkException.getMessage().contains("Executor heartbeat timed out") ||
+                                sparkException.getMessage().contains("Unable to talk to the executor")) {
```

Comment:
@arhimondr : OSS Spark use "Executor heartbeat timed out". However, when fb-spark backportshttps://github.com/apache/spark/commit/b5c35d68e430d6ebf92aff9f75f730ca14a68469. they use a slightly different exception message "Unable to talk to the executor". 

But seems both are quite self-explained, I feel no hurt to use both of them.